holopy/scattering/theory/scatteringtheory.py

```diff
@@ -16,220 +16,62 @@
 # You should have received a copy of the GNU General Public License
 # along with HoloPy.  If not, see <http://www.gnu.org/licenses/>.
 """
-Base class for scattering theories.  Implements python-based
-calc_intensity and calc_holo, based on subclass's calc_field
 .. moduleauthor:: Jerome Fung <jerome.fung@post.harvard.edu>
 .. moduleauthor:: Vinothan N. Manoharan <vnm@seas.harvard.edu>
 .. moduleauthor:: Thomas G. Dimiduk <tdimiduk@physics.harvard.edu>
 .. moduleauthor:: Brian Leahy <bleahy@g.harvard.edu>
 """
-
-from warnings import warn
-
 import numpy as np
-import xarray as xr
 
-from holopy.core.math import find_transformation_function
 from holopy.core.holopy_object import HoloPyObject
-from holopy.scattering.scatterer import Scatterers
-from holopy.scattering.errors import TheoryNotCompatibleError, MissingParameter
-from holopy.core.metadata import (
-    vector, illumination, flat, update_metadata, clean_concat)
-from holopy.core.utils import ensure_array
 try:
     from holopy.scattering.theory.mie_f import mieangfuncs
 except ImportError:
     pass
 
 
-def get_wavevec_from(schema):
-    return 2 * np.pi / (schema.illum_wavelen / schema.medium_index)
-
-
 class ScatteringTheory(HoloPyObject):
     """
     Defines common interface for all scattering theories.
 
+    Subclasses must implement:
+    * can_handle
+    * raw_fields or raw_scat_matrs or both.
+    * (optionally) raw_cross_sections,
+
     Notes
     -----
-    A subclasses that do the work of computing scattering should do it
-    by implementing _raw_fields and/or _raw_scat_matrs and (optionally)
-    _raw_cross_sections. _raw_cross_sections is needed only for
-    calc_cross_sections. Either of _raw_fields or _raw_scat_matrs will
-    give you calc_holo, calc_field, and calc_intensity. Obviously
-    calc_scat_matrix will only work if you implement _raw_cross_sections.
-    So the simplest thing is to just implement _raw_scat_matrs. You only
-    need to do _raw_fields there is a way to compute it more efficently
-    and you care about that speed, or if it is easier and you don't care
-    about matrices.
+    Subclasses should implement the following methods to create a
+    scatteringtheory that works for the following user-facing methods:
+    * `calc_holo`:              `raw_fields` or `raw_scat_matrs`
+    * `calc_intensity`:         `raw_fields` or `raw_scat_matrs`
+    * `calc_field`:             `raw_fields` or `raw_scat_matrs`
+    * `calc_scat_matrix`:       `raw_scat_matrs`
+    * `calc_cross_sections`:    `raw_cross_sections`
+
+    By default, ScatteringTheories computer `raw_fields` from the
+    `raw_scat_matrs`; over-ride the `raw_fields` method to compute the
+    fields in a different way.
```

Comment:
This write-up is much more clear than what was here previously!

Reply:
Thanks!